tests/ui/unbound_return_lifetimes.rs

```diff
@@ -0,0 +1,74 @@
+#![allow(unused, dead_code, clippy::needless_lifetimes)]
+#![warn(clippy::unbound_return_lifetimes)]
+
+use std::collections::HashMap;
+use std::hash::Hash;
+
+struct FooStr(str);
+
+fn unbound_fun<'a>(x: impl AsRef<str> + 'a) -> &'a FooStr {
```

Comment:
I'd like to see a test with a `&'static _` return type, to rule out false positives.